src/components/utils/icon.ts

```diff
@@ -19,6 +19,20 @@ export function fieldKindToColor(
   return fieldKindMap[kind];
 }
 
+export function fieldToIcon(field: Malloy.FieldInfo): IconType | undefined {
+  switch (field.kind) {
+    case 'view':
+      return 'query';
+    case 'dimension':
+    case 'measure':
+      return atomicTypeToIcon(field.type.kind);
+    case 'join':
+      return relationshipToIcon(field.relationship);
+    default:
+      return undefined;
+  }
+}
```

Comment:
It doesn't matter a whole lot, and it seems like a TypeScript bug, but if you re-write it like this you have no `undefined` results.
```suggestion
export function fieldToIcon(field: Malloy.FieldInfo): IconType {
  let icon: IconType;
  switch (field.kind) {
    case 'view':
      icon = 'query';
      break;
    case 'dimension':
    case 'measure':
      icon = atomicTypeToIcon(field.type.kind);
      break;
    case 'join':
      icon = relationshipToIcon(field.relationship);
      break;
  }
  return icon;
}
```